internal/store/fleet.go

```diff
@@ -208,6 +210,10 @@ func (s *FleetStore) createOrUpdateTx(tx *gorm.DB, ctx context.Context, orgId uu
 				fleet.Generation = existingRecord.Generation
 			}
 
+			if !sameSelector {
+				selectorUpdated = true
```

Comment:
some useless comment ahead:
it could be this, but probably less idiomatic. `selectorUpdated = selectorUpdated || !sameSelector`